File: sistema-reclamos-backend/src/main/java/com/reclamos/entities/Reclamo.java
package com.reclamos.entities;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.CurrentTimestamp;

import com.reclamos.enums.ReclamoEstado;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Entity
@Table(name = "reclamo")
@AllArgsConstructor
@NoArgsConstructor
@Getter @Setter
@Builder
public class Reclamo {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @CreationTimestamp
  private LocalDateTime fechaRegistro;
  private String tipoDocumento;
  private String numeroDocumento;
  private String nombres;
  private String apellidos;
  private String direccion;
  private String telefono;
  private String email;
  private String tipoBien;
  private BigDecimal montoReclamado;
  private String descripcionReclamo;
  private String pedidoConsumidor;

  @Enumerated(EnumType.STRING)
  @Builder.Default
  private ReclamoEstado estado = ReclamoEstado.PENDIENTE;

  private LocalDateTime fechaCierre;
}

File: sistema-reclamos-frontend/src/types/reclamo.ts
export interface Reclamo {
  id: number;
  tipoDocumento: string;
  numeroDocumento: string;
  nombres: string;
  apellidos: string;
  direccion: string;
  telefono: string;
  email: string;
  tipoBien: string;
  montoReclamado: number;
  descripcionReclamo: string;
  pedidoConsumidor: string;
  estado?: ReclamoEstado;
  fechaRegistro?: string;
  fechaCierre?: string | null;
}

export type ReclamoEstado = 'PENDIENTE' | 'EN_PROCESO' | 'ATENDIDO' | 'CERRADO';

export interface CreateReclamoDto {
  tipoDocumento: string;
  numeroDocumento: string;
  nombres: string;
  apellidos: string;
  direccion: string;
  telefono: string;
  email: string;
  tipoBien: string;
  montoReclamado: number;
  descripcionReclamo: string;
  pedidoConsumidor: string;
}

export interface UpdateEstadoDto {
  reclamoEstado: ReclamoEstado;
}
